### PR TITLE
WIP: on top of muxer, add renames to ProviderMeta

### PR DESCRIPTION
**Pull request by t0yv0**
@iwahbe early preview. I got this to pass tests but go embed limitations require some file motion/movement.

The change makes Plugin Framework providers (even unmuxed) to write Renames into MetadataInfo, which is now required for PF providers.

This imposes a change on the user code, unfortunately, so that will need to be called out in release notes. 

Also looks like I need to update pf/README.md examples for this.

### CODE REVIEW

File: pkg/tfgen/generate.go [hunk 1]
```diff
@@ -1800,3 +1803,20 @@ func ignoreMappingError(s []string, str string) bool {
 	}
 	return false
 }
+
+func addRenamesToMetadataInfo(g *Generator) error {
```
Comment (t0yv0) [inline pkg/tfgen/generate.go:1807]: This has the side-effect of adding renames to providers that don't strictly need them, but opt into MetadataInfo.

**Review by iwahbe — APPROVED** [review]
I think this is better. Its breaking, but future adoptions only need to embed one file, not multiple files.

**Comment by t0yv0** [comment]
Yeah I think it's the right direction but I itch to rename ProviderMetadata; later.. OK I will rebase on master.

**Comment by t0yv0** [comment]
Closing in fav of https://github.com/pulumi/pulumi-terraform-bridge/pull/964 cleaned up and rebased on master.

**Pull request closed without merging**